source/server/hot_restarting_child.cc

```diff
@@ -7,6 +7,43 @@ namespace Server {
 
 using HotRestartMessage = envoy::HotRestartMessage;
 
+void HotRestartingChild::UdpForwardingContext::registerListener(
+    Network::Address::InstanceConstSharedPtr address,
+    std::shared_ptr<Network::UdpListenerConfig> listener_config) {
+  const bool inserted =
+      listener_map_.try_emplace(address->asString(), ForwardEntry{address, listener_config}).second;
+  ASSERT(inserted, "Two udp listeners on the same address shouldn't be possible");
```

Comment:
Out of curiosity, what enforces the invariant that there only is a single listener per interface?

Reply:
Per interface *and port*. I'm not sure if envoy enforces that you can't, and SO_REUSEPORT means the operating system won't enforce that you can't, but I *hope* envoy enforces that you can't because if you do have multiple listeners with different behaviors on the same interface and port you'd get horrific mixed delivery. (And if you have multiple listeners on the same interface+port with identical behavior then why?!)